src/unitxt/metrics.py

```diff
@@ -3181,3 +3182,90 @@ def compute(
                 best_thr = thr
 
         return {self.main_score: best_acc, "best_thr_max_acc": best_thr}
+
+
+######################
+# RerankRecallMetric #
+
+def pytrec_eval_at_k(results, qrels, at_k, metric_name):
+    metric = {}
+
+    for k in at_k:
+        metric[f"{metric_name}@{k}"] = 0.0
+
+    metric_string = f"{metric_name}." + ",".join([str(k) for k in at_k])
+    # print('metric_string = ', metric_string)
+    evaluator = pytrec_eval.RelevanceEvaluator(
+        qrels, {"ndcg", metric_string}
+    )  # {map_string, ndcg_string, recall_string, precision_string})
+    scores = evaluator.evaluate(results)
+    scores = pd.DataFrame(scores).transpose()
+
+    keys = []
+    column_map = {}
+    for k in at_k:
+        keys.append(f"{metric_name}_{k}")
+        column_map[f"{metric_name}_{k}"] = k
+    scores[keys].rename(columns=column_map)
+
+    return scores
+
+
+class RerankRecall(GlobalMetric):
+    """RerankRecall: measures the quality of reranking with respect to ground truth ranking scores.
+
+    This metric measures ranking performance across a dataset.  The
+    references for a query will have a score of 1 for the gold passage
+    and 0 for all other passages.  The model returns scores in [0,1]
+    for each passage,query pair.  This metric measures recall at k by
+    testing that the predicted score for the gold passage,query pair
+    is at least the k'th highest for all passages for that query.  A
+    query receives 1 if so, and 0 if not.  The 1's and 0's are
+    averaged across the dataset.
+
+    query_id_field selects the field containing the query id for an instance.
+    passage_id_field selects the field containing the passage id for an instance.
+    at_k selects the value of k used to compute recall.
+
+    """
+    main_score = "recall_at_5"
+    query_id_field: str = "query_id"
+    passage_id_field: str = "passage_id"
+    at_k: List[int] = [1, 2, 5]
+
```

Comment:
```suggestion
     _requirements_list: List[str] = ["pytrec_eval"]
```
Unitxt by default does not operate with requirements but each operator/metric decalre its requirements

Reply:
done